middle_end/flambda/types/template/flambda_type.templ.ml

```diff
@@ -741,7 +741,146 @@ let prove_strings env t : String_info.Set.t proof =
   | Naked_immediate _ | Naked_float _ | Naked_int32 _ | Naked_int64 _
   | Naked_nativeint _ -> wrong_kind ()
 
-let prove_block_field_simple env ~min_name_mode t field_index : Simple.t proof =
+let prove_is_always_tagging_of_simple env ~min_name_mode t : Simple.t proof =
```

Comment:
Those functions are too similar not to factorise them. Otherwise it is quite hard to tell the actual diference

Reply:
I've factored them in the most recent commits.